engine/lite/components/kafka-tests/src/test/scala/pl/touk/nussknacker/engine/lite/components/UniversalSourceJsonFunctionalTest.scala

```diff
@@ -12,60 +12,81 @@ import org.scalatest.matchers.should.Matchers
 import org.scalatest.prop.TableDrivenPropertyChecks
 import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
 import pl.touk.nussknacker.engine.api.CirceUtil
-import pl.touk.nussknacker.engine.api.typed.CustomNodeValidationException
 import pl.touk.nussknacker.engine.api.validation.ValidationMode
 import pl.touk.nussknacker.engine.build.ScenarioBuilder
 import pl.touk.nussknacker.engine.canonicalgraph.CanonicalProcess
-import pl.touk.nussknacker.engine.lite.components.utils.JsonTestData.{InputEmptyObject, schemaObjNull, _}
-import pl.touk.nussknacker.engine.lite.util.test.{KafkaConsumerRecord, LiteKafkaTestScenarioRunner}
+import pl.touk.nussknacker.engine.lite.components.utils.JsonTestData._
+import pl.touk.nussknacker.engine.lite.util.test.KafkaConsumerRecord
 import pl.touk.nussknacker.engine.schemedkafka.KafkaUniversalComponentTransformer._
 import pl.touk.nussknacker.engine.schemedkafka.schemaregistry.SchemaVersionOption
 import pl.touk.nussknacker.engine.util.test.TestScenarioRunner.RunnerListResult
-import pl.touk.nussknacker.engine.util.test.{RunListResult, RunResult, TestScenarioRunner}
+import pl.touk.nussknacker.engine.util.test.{RunListResult, RunResult}
 import pl.touk.nussknacker.test.{SpecialSpELElement, ValidatedValuesDetailedMessage}
 
-import java.util.UUID
-
 class UniversalSourceJsonFunctionalTest extends AnyFunSuite with Matchers with ScalaCheckDrivenPropertyChecks with Inside
-  with TableDrivenPropertyChecks with ValidatedValuesDetailedMessage {
+  with TableDrivenPropertyChecks with ValidatedValuesDetailedMessage with FunctionalTestMixin {
 
-  import pl.touk.nussknacker.test.LiteralSpELImplicits._
   import LiteKafkaComponentProvider._
-  import LiteKafkaTestScenarioRunner._
   import SpecialSpELElement._
   import pl.touk.nussknacker.engine.spel.Implicits._
+  import pl.touk.nussknacker.test.LiteralSpELImplicits._
+
+  test("should test end to end kafka json data at sink and source / handling nulls and empty json" ) {
+    val testData = Table(
+      ("config", "result"),
+      (oConfig(InputEmptyObject, schemaObjNull, schemaObjNull), valid(obj())),
+      (oConfig(InputEmptyObject, schemaObjNull, schemaObjNull, OutputField), oValid(Null)),
+      (oConfig(Null, schemaObjNull, schemaObjNull), oValid(Null)),
+      (oConfig(Null, schemaObjNull, schemaObjNull, OutputField), oValid(Null)),
 
-  private val sourceName = "my-source"
-  private val sinkName = "my-sink"
+      (oConfig(InputEmptyObject, schemaObjString, schemaObjString), valid(obj())),
+      (oConfig(InputEmptyObject, schemaObjString, schemaObjString, OutputField), valid(obj())), //FIXME: it should throw exception at runtime
```

Comment:
why it should throw?

Reply:
Because you want to pass explicitly null to string schema. `OutputField` returns explicitly null.